lib/redcord/migration/index.rb

```diff
@@ -0,0 +1,66 @@
+# frozen_string_literal: true
+
+# typed: strict
+
+module Redcord::Migration::Index
+  extend T::Sig
+
+  sig { params(model: T.class_of(Redcord::Base), index_name: Symbol).void }
+  def remove_index(model, index_name)
+    if model.redis.sismember("#{model.model_key}:index_attrs", index_name)
+      _remove_inex_from_attr_set(
+        model: model,
+        attr_set_name: 'index_attrs',
+        index_name: index_name,
+      )
+
+      model.redis.scan_each(match: "#{model.model_key}:#{index_name}:*") { |key| _del_set(model, key) }
+    elsif model.redis.sismember("#{model.model_key}:range_index_attrs", index_name)
+      _remove_inex_from_attr_set(
+        model: model,
+        attr_set_name: 'range_index_attrs',
+        index_name: index_name,
+      )
+
+      model.redis.scan_each(match: "#{model.model_key}:#{index_name}:") { |key| _del_set(model, key) }
+      model.redis.scan_each(match: "#{model.model_key}:#{index_name}:") { |key| model.redis.del(key) }
+      model.redis.scan_each(match: "#{model.model_key}:#{index_name}") { |key| _del_zset(model, key) }
```

Comment:
We can use `zscan_each` here as well (unless there's some inclination to use `scan` instead.

Reply:
similar as above